assets/src/components/Sidebar/NewThemeSettings.js

```diff
@@ -0,0 +1,87 @@
+import { SelectControl } from '@wordpress/components';
+import { useState, useEffect } from 'react';
+import { p4ServerThemes } from '../../theme/p4ServerThemes';
+import { __ } from '@wordpress/i18n';
+import { useDispatch } from '@wordpress/data';
+
+const keysAsLabel = obj => Object.keys(obj).map(k => ({ label: k, value: k }));
+
+// Just a lightweight way to have a separate UI element in the sidebar so that it's clear which themes are legacy and
+// which are new. This is likely just for internal usage to facilitate reviewing the refactored version, eventually
+// editors will work with a single select.
+const useServerThemes = () => {
+  const [serverThemes, setServerThemes] = useState({});
+
+  useEffect(() => {
+    const fetchThemes = async () => {
+      const themes = await p4ServerThemes.fetchThemes();
+      setServerThemes(themes);
+    };
+    fetchThemes();
+  }, []);
+
+  return serverThemes;
+};
+
+const getAllDefinedProps = () => Object.entries(document.documentElement.style).filter(([, k]) => {
```

Comment:
I'll change this to `Object.values`

Reply:
`document.documentElement.style` is somewhat weird. For custom properties, it returns integer keys, and the value is the property name. It doesn't give you the custom property value. For regular properties of the object it acts normally.

I'll add an explanatory comment to the code.